gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -256,6 +255,23 @@ public static SerDeInfo getSerDeInfo(HiveRegistrationUnit unit) {
     return si;
   }
 
+  /**
+   * Util for detecting if a table has a complex union (aka non-optional unions) column types.
+   *
+   * @param t
+   * @return if hive table contains complex uniontype columns
+   */
+  public static boolean containsUnionTypeColumn(HiveTable t) {
+    return t.getColumns().stream()
+        .map(HiveRegistrationUnit.Column::getType)
+        .map(Object::toString)
+        .anyMatch(columnType -> columnType.contains("uniontype"));
```

Comment:
Seems to me we only check union type but not non-optional union type. Is this intentional? If so, please update the description to reflect that as well.